packages/ui/src/elements/WhereBuilder/reduceFieldMap.tsx

```diff
@@ -1,38 +1,109 @@
 
-export const reduceFieldMap = (
-  fieldMap: FieldMap,
-  i18n,
-  labelPrefix?: string,
-  pathPrefix?: string,
-  locale?: string,
-) => {
+export type ReduceFieldMapArgs = {
+  fieldMap: FieldMap
+  i18n: I18nClient
+  labelPrefix?: string
+  pathPrefix?: string
+}
+
+/**
+ * Reduces a field map to a flat array of fields with labels and values.
+ * Used in the WhereBuilder component to render the fields in the dropdown.
+ */
+export const reduceFieldMap = ({ fieldMap, i18n, labelPrefix, pathPrefix }: ReduceFieldMapArgs) => {
   return fieldMap.reduce((reduced, field) => {
     if (field.disableListFilter) return reduced
 
     if (field.type === 'tabs' && 'tabs' in field.fieldComponentProps) {
       const tabs = field.fieldComponentProps.tabs
       tabs.forEach((tab) => {
-        if (tab.name && typeof tab.label === 'string' && tab.fieldMap) {
-          reduced.push(...reduceFieldMap(tab.fieldMap, i18n, tab.label, tab.name))
+        if (typeof tab.label !== 'boolean') {
+          const localizedTabLabel = getTranslation(tab.label, i18n)
+          const labelWithPrefix = labelPrefix
+            ? labelPrefix + ' > ' + localizedTabLabel
+            : localizedTabLabel
+
+          // Make sure we handle nested tabs
+          const tabPathPrefix = tab.name
+            ? pathPrefix
+              ? pathPrefix + '.' + tab.name
+              : tab.name
+            : pathPrefix
+
+          if (typeof localizedTabLabel === 'string') {
+            reduced.push(
+              ...reduceFieldMap({
+                fieldMap: tab.fieldMap,
+                i18n,
+                labelPrefix: labelWithPrefix,
+                pathPrefix: tabPathPrefix,
+              }),
+            )
+          }
         }
       })
       return reduced
     }
 
+    // Rows cant have labels, so we need to handle them differently
+    if (field.type === 'row' && 'fieldMap' in field.fieldComponentProps) {
+      reduced.push(
+        ...reduceFieldMap({
+          fieldMap: field.fieldComponentProps.fieldMap,
+          i18n,
+          labelPrefix,
+          pathPrefix,
+        }),
+      )
+      return reduced
+    }
+
+    if (field.type === 'collapsible' && 'fieldMap' in field.fieldComponentProps) {
+      const localizedTabLabel = getTranslation(field.fieldComponentProps.label, i18n)
+      const labelWithPrefix = labelPrefix
+        ? labelPrefix + ' > ' + localizedTabLabel
+        : localizedTabLabel
+
+      reduced.push(
+        ...reduceFieldMap({
+          fieldMap: field.fieldComponentProps.fieldMap,
+          i18n,
+          labelPrefix: labelWithPrefix,
+          pathPrefix,
+        }),
+      )
+      return reduced
+    }
+
     if (field.type === 'group' && 'fieldMap' in field.fieldComponentProps) {
+      const translatedLabel = getTranslation(field.fieldComponentProps.label, i18n)
+
+      const labelWithPrefix = labelPrefix ? labelPrefix + ' > ' + translatedLabel : translatedLabel
```

Comment:
This does not consider the scenario where labelPrefix is set (we are already in a nested group), but the current group does not have any label (translatedLabel is undefined).

```suggestion
      const labelWithPrefix = labelPrefix
        ? translatedLabel
          ? labelPrefix + ' > ' + translatedLabel
          : labelPrefix
        : translatedLabel
```